UnityProject/Assets/Scripts/Managers/PlayerList.cs

```diff
@@ -140,6 +140,17 @@ public List<ConnectedPlayer> GetPlayersOnMatrix(MatrixInfo matrix)
 		return InGamePlayers.FindAll(p => (p.Script != null) && p.Script.registerTile.Matrix.Id == matrix?.Id);
 	}
 
+	public ConnectedPlayer GetPlayerByID(string id)
+	{
+		foreach (var player in AllPlayers)
+		{
+			if(player.UserId != id) continue;
```

Comment:
if you invert this, you won't need the ``continue`` making the algorithm have less exit points